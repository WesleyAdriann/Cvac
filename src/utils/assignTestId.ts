import { Platform } from 'react-native'

interface ITestID {
  accessible?: boolean
  testID: string
  accessibilityLabel?: string
}

type TComponentType =
  | 'View'
  | 'ScrollView'
  | 'TextInput'
  | 'Text'
  | 'SafeAreaView'
  | 'Pressable'
  | 'Modal'

export const assignTestId = (
  componentType: TComponentType,
  componentId: string
): ITestID => {
  const id = `${componentType}_${componentId}`
  return Platform.OS === 'android'
    ? {
        testID: id,
        accessible: true,
        accessibilityLabel: id
      }
    : {
        testID: id,
        accessible: false
      }
}
export const mockTestID = (
  componentType: TComponentType,
  componentId: string
): ITestID => assignTestId(componentType, componentId)
