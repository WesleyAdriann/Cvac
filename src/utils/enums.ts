export enum ECalendarsName {
  'kid' = 'Crian√ßa',
  'elder' = 'Idoso',
  'adult' = 'Adulto',
  'teen' = 'Adolescente'
}
