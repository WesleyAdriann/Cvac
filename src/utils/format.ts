export const formatDate = (value: string) => {
  const clearDate = value.replace(/\D/g, '')

  if (clearDate.length === 8) return value

  const day = clearDate.slice(0, 2)
  const month = clearDate.slice(2, 4)
  const year = clearDate.slice(4, 8)

  if (!month) return day
  if (!year) return `${day}/${month}`
  return `${day}/${month}/${year}`
}
