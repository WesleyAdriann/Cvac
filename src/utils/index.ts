export * from './assignTestId'
export * from './withProviders'
export * from './format'
export * from './logs'
