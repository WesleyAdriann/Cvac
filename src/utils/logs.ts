export const log = (TAG: string, ...args: any[]) => {
  if (__DEV__) console.log(`[${TAG}] `, ...args)
}
