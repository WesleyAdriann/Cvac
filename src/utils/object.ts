export const isEmpty = (obj: object | null) => {
  if (!obj) return true
  // eslint-disable-next-line no-unreachable-loop
  for (const key in obj) return false
  return true
}
