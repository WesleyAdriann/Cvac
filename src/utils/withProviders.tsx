import React from 'react'
import { Provider as PaperProvider } from 'react-native-paper'
import { ThemeProvider } from 'styled-components'

import { theme, paperTheme } from '../tokens'

export const TestsProviders: React.FC = ({ children }) => (
  <PaperProvider theme={paperTheme} >
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </PaperProvider>
)
