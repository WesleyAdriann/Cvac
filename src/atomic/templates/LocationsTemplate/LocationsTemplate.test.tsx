
import React from 'react'
import { render, cleanup } from '@testing-library/react-native'

import { AllTheProviders as wrapper, mockTestID } from '../../../utils'

import { LocationsTemplate, ILocationsTemplate } from './index'

describe('Templates/LocationsTemplate', () => {
  const props: ILocationsTemplate = {
    testID: 'LocationsTemplate',
    mapRegion: { latitude: 0, latitudeDelta: 0, longitude: 0, longitudeDelta: 0 },
    onPressLocation: jest.fn(),
    rangeInput: '3',
    onChangeRange: jest.fn()
  }

  const elContainer = mockTestID('SafeAreaView', props.testID!).testID

  const renderComponent = () => render(<LocationsTemplate {...props} />, { wrapper })
  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  it(`should render component #${elContainer}`, async () => {
    const sut = await component.findByTestId(elContainer)
    expect(sut).toBeTruthy()
  })

  it('should render snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
