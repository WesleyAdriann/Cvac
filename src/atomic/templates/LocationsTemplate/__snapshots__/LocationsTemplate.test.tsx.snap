// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/LocationsTemplate should render snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    accessible={false}
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
        Object {
          "backgroundColor": "#fff",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
    testID="SafeAreaView_LocationsTemplate"
  >
    <View
      accessible={false}
      flex={1}
      paddingStyle={0}
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
        ]
      }
      testID="View_Flex"
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <View
          accessible={false}
          flex={1}
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexDirection": "column",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 0,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
          testID="View_Flex"
        >
          <View
            accessible={false}
            flex={1}
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexDirection": "column",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
            testID="View_Map"
          >
            <AIRMap
              handlePanDrag={false}
              initialRegion={null}
              onChange={[Function]}
              onLayout={[Function]}
              onMapReady={[Function]}
              onMarkerPress={[Function]}
              region={
                Object {
                  "latitude": 0,
                  "latitudeDelta": 0,
                  "longitude": 0,
                  "longitudeDelta": 0,
                }
              }
              showsUserLocation={true}
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          accessible={false}
          style={
            Array [
              Object {
                "flexDirection": "column",
                "marginBottom": 0,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
          testID="View_Flex"
        >
          <RCTScrollView
            accessible={false}
            contentContainerStyle={
              Object {
                "flexGrow": 1,
                "padding": undefined,
              }
            }
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
                Object {
                  "maxHeight": 144,
                },
              ]
            }
            testID="ScrollView_Flex"
          >
            <View />
          </RCTScrollView>
          <View
            accessible={false}
            paddingStyle="16px 16px 0"
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                },
              ]
            }
            testID="View_Flex"
          >
            <View
              accessible={false}
              style={
                Array [
                  Object {
                    "flexDirection": "column",
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                  },
                ]
              }
              testID="View_TextInput"
            >
              <View
                style={Object {}}
              >
                <View>
                  <View
                    pointerEvents="none"
                    style={
                      Array [
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 6,
                        },
                        Object {
                          "backgroundColor": "#fff",
                          "borderColor": "#0000001f",
                          "borderRadius": 4,
                          "borderWidth": 1,
                        },
                      ]
                    }
                  />
                  <View
                    style={
                      Array [
                        Object {
                          "paddingBottom": 0,
                        },
                        Object {
                          "minHeight": 64,
                          "paddingTop": 8,
                        },
                      ]
                    }
                  >
                    <View
                      collapsable={false}
                      nativeID="animatedComponent"
                      pointerEvents="none"
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "opacity": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "transform": Array [
                            Object {
                              "translateX": 3,
                            },
                          ],
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        nativeID="animatedComponent"
                        pointerEvents="none"
                        style={
                          Object {
                            "backgroundColor": "#fff",
                            "bottom": 4,
                            "left": 10,
                            "opacity": 1,
                            "position": "absolute",
                            "right": 0,
                            "top": 6,
                            "transform": Array [
                              Object {
                                "translateX": -3,
                              },
                            ],
                            "width": 8,
                          }
                        }
                      />
                      <Text
                        collapsable={false}
                        nativeID="animatedComponent"
                        numberOfLines={1}
                        style={
                          Object {
                            "backgroundColor": "#fff",
                            "color": "transparent",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": undefined,
                            "left": 18,
                            "opacity": 1,
                            "paddingHorizontal": 0,
                            "position": "absolute",
                            "textAlign": "left",
                            "top": 37,
                            "transform": Array [
                              Object {
                                "translateX": 0,
                              },
                              Object {
                                "translateY": -30,
                              },
                              Object {
                                "scale": 0.75,
                              },
                              Object {
                                "scaleY": 0.2,
                              },
                            ],
                            "writingDirection": "ltr",
                          }
                        }
                      >
                        Distância máxima
                      </Text>
                      <Text
                        collapsable={false}
                        nativeID="animatedComponent"
                        numberOfLines={1}
                        onLayout={[Function]}
                        style={
                          Object {
                            "color": "#00C9D2",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": undefined,
                            "left": 0,
                            "opacity": 0,
                            "paddingHorizontal": 14,
                            "position": "absolute",
                            "textAlign": "left",
                            "top": 36,
                            "transform": Array [
                              Object {
                                "translateX": 0,
                              },
                              Object {
                                "translateY": -30,
                              },
                              Object {
                                "scale": 0.75,
                              },
                            ],
                            "writingDirection": "ltr",
                          }
                        }
                      >
                        Distância máxima
                      </Text>
                      <Text
                        collapsable={false}
                        nativeID="animatedComponent"
                        numberOfLines={1}
                        style={
                          Object {
                            "color": "rgba(0, 0, 0, 0.54)",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": undefined,
                            "left": 0,
                            "opacity": 0,
                            "paddingHorizontal": 14,
                            "position": "absolute",
                            "textAlign": "left",
                            "top": 36,
                            "transform": Array [
                              Object {
                                "translateX": 0,
                              },
                              Object {
                                "translateY": -30,
                              },
                              Object {
                                "scale": 0.75,
                              },
                            ],
                            "writingDirection": "ltr",
                          }
                        }
                      >
                        Distância máxima
                      </Text>
                    </View>
                    <TextInput
                      accessible={false}
                      autoCapitalize="sentences"
                      autoCorrect={true}
                      editable={true}
                      keyboardType="number-pad"
                      multiline={false}
                      onBlur={[Function]}
                      onChangeText={[Function]}
                      onFocus={[Function]}
                      placeholder=""
                      placeholderTextColor="rgba(0, 0, 0, 0.54)"
                      secureTextEntry={false}
                      selectionColor="#00C9D2"
                      style={
                        Array [
                          Object {
                            "flexGrow": 1,
                            "margin": 0,
                            "paddingHorizontal": 14,
                            "zIndex": 1,
                          },
                          Object {
                            "height": 56,
                          },
                          Object {
                            "paddingBottom": 0,
                            "paddingTop": 0,
                          },
                          Object {
                            "color": "#424242",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": undefined,
                            "textAlign": "left",
                            "textAlignVertical": "center",
                          },
                          false,
                          Object {
                            "marginRight": 0,
                            "paddingRight": 44,
                          },
                        ]
                      }
                      testID="TextInput_TextInput"
                      underlineColorAndroid="transparent"
                      value="3"
                    />
                  </View>
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    onLayout={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "opacity": 1,
                        "position": "absolute",
                        "right": 12,
                        "top": 35,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "rgba(66, 66, 66, 0.54)",
                          },
                          Object {
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": undefined,
                          },
                          undefined,
                        ]
                      }
                    >
                      km
                    </Text>
                  </View>
                </View>
              </View>
              <Text
                accessible={false}
                collapsable={false}
                nativeID="animatedComponent"
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgba(66, 66, 66, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 12,
                    "fontWeight": "400",
                    "opacity": 0,
                    "paddingHorizontal": 12,
                    "paddingVertical": 4,
                    "textAlign": "left",
                    "transform": Array [],
                    "writingDirection": "ltr",
                  }
                }
                testID="Text_TextInput_helper"
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
